src/test/java/com/ot/restaurant/repositories/CustomerRepositoryImplTest.java

```diff
@@ -0,0 +1,89 @@
+package com.ot.restaurant.repositories;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.ot.restaurant.entities.Customer;
+import com.ot.restaurant.exceptions.IdNullException;
+import constants.Status;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class CustomerRepositoryImplTest {
+  private CustomerRepositoryImpl customerRepository = new CustomerRepositoryImpl();
+  private List<Customer> customerListTest = new ArrayList<>();
+  private final String firstName = "Loki1";
+  private final String secondName = "Loki2";
+  private final String thirdName = "Loki3";
+
+  private Customer createCustomer(String firstName) {
+    Customer customer = new Customer();
+    customer.setFirstName(firstName);
+    customer.setStatus(Status.ACTIVE);
+    return customer;
+  }
+
+  @BeforeEach
+  void setUp() throws Exception {
+    customerListTest.add(createCustomer("Loki0"));
+    customerRepository.insert(customerListTest.get(0));
+    customerListTest.add(createCustomer("Loki1"));
+    customerRepository.insert(customerListTest.get(1));
+    customerListTest.add(createCustomer("Loki2"));
+    customerRepository.insert(customerListTest.get(2));
```

Comment:
I think you could do something like this:
```
final var customers = List.of(
CustumerFixtures.buildCustomerFromExample(Customer.builder().withName("test1").build()),
CustumerFixtures.buildCustomerFromExample(Customer.builder().withName("test2").build()),
CustumerFixtures.buildCustomerFromExample(Customer.builder().withName("test3").build()),
CustumerFixtures.buildCustomerFromExample(Customer.builder().withName("test4").withStatus(Status.DELETED).build()),
CustumerFixtures.buildCustomerFromExample(Customer.builder().withName("test5").withStatus(Status.DELETED).build()),
);

customers.forEach(customerRepository::add);
```

Reply:
Probably you will need to do this at class level to make it accessible for other methods